test/dashboard-e2e/api/api-helpers.js

```diff
@@ -7,51 +7,82 @@ export class ApiHelpers {
         this.bearerToken = bearerToken;
     }
 
-    async getTests() {
-        const request = `${this.apiUrl}/tests`;
-
+    async makeGet(requestUrl) {
         try {
-            const response = await superagent.get(request)
+            const response = await superagent.get(requestUrl)
                 .set('Content-Type', 'application/json')
                 .set('Authorization', this.cloudContext ? `Bearer ${this.bearerToken}` : '');
 
             return ApiHelpers.parseResponse(response);
             
         } catch (e) {
-            throw Error(`getTests failed on "${request}" with: "${e}"`);
+            throw Error(`makeGet failed on "${requestUrl}" with: "${e}"`);
         }
     }
 
-    async getTestSuites() {
-        const request = `${this.apiUrl}/test-suites`;
-
+    async makePost(requestUrl, requestData) {
         try {
-            const response = await superagent.get(request)
+            const response = await superagent.post(requestUrl)
                 .set('Content-Type', 'application/json')
-                .set('Authorization', this.cloudContext ? `Bearer ${this.bearerToken}` : '');
+                .set('Authorization', this.cloudContext ? `Bearer ${this.bearerToken}` : '')
+                .send(requestData);
 
             return ApiHelpers.parseResponse(response);
-            
         } catch (e) {
-            throw Error(`getTestSuites failed on "${request}" with: "${e}"`);
+            throw Error(`createTest failed on "${requestUrl}" with: "${e}"`);
         }
     }
 
-    async createTest(testData) {
-        const request = `${this.apiUrl}/tests`;
-        
+    async makePatch(requestUrl, requestData) {
         try {
-            const response = await superagent.post(request)
+            const response = await superagent.patch(requestUrl)
                 .set('Content-Type', 'application/json')
                 .set('Authorization', this.cloudContext ? `Bearer ${this.bearerToken}` : '')
-                .send(testData);
+                .send(requestData);
 
             return ApiHelpers.parseResponse(response);
         } catch (e) {
-            throw Error(`createTest failed on "${request}" with: "${e}"`);
+            throw Error(`updateTest failed on "${requestUrl}" with: "${e}"`);
         }
     }
 
+    async makeDelete(requestUrl) {
+        try {
+            await superagent.delete(requestUrl)
+                .set('Authorization', this.cloudContext ? `Bearer ${this.bearerToken}` : '');
+        } catch (e) {
+            throw Error(`removeTest failed on "${requestUrl}" with: "${e}"`);
+        }
+    }
+
+    async getTests() {
+        const requestUrl = `${this.apiUrl}/tests`;
+        const response = await this.makeGet(requestUrl);
+
+        return response;
```

Comment:
```suggestion
        return this.makeGet(`${this.apiUrl}/tests`);
```

and you may do similarly for all others :)

Reply:
Isn't await needed here? ESlint complained about `return await this.makeGet(requestUrl);` - that's why I added additional response const.

Reply:
`await` is not needed here - `this.makeGet` is an async function, so it will return a promise.